docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,304 @@
+`module`  | String | The module name. TestCafe searches for the module's entry point with Node.js machinery and injects its content into the tested page.
+
+**Examples**
+
+```sh
+testcafe chrome my-tests --cs lodash
+```
+
+```js
+fixture `My fixture`
+    .page `https://example.com`
+    .clientScripts('lodash');
+
+// or
+
+fixture `My fixture`
+    .page `https://example.com`
+    .clientScripts({ module: 'lodash' });
+```
+
+```js
+{
+    "clientScripts": "lodash",
+    // or
+    "clientScripts": { "module": "lodash" }
+}
+```
+
+> You cannot combine the `module`, [path](#specify-the-path-to-a-javascript-file) and [content](#specify-the-script-code) properties.
+
+### Specify the Script Code
+
+You can provide the injected script as a string with JavaScript code. Pass an object with the `content` property to do this.
+
+*Programming interface*
+
+```text
+runner.clientScripts({ content })
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": { content }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts({ content })
+test.clientScripts({ content })
+```
+
+Argument  | Type   | Description
+--------- | ------ | ----------------------------------------
+`content` | String | JavaScript code that should be injected.
+
+**Examples**
+
+```js
+const mockDate = `
+    Date.prototype.getTime = function () {
+        return 42;
+    };
+`;
+
+test
+    ('My test', async t => { /* ... */ })
+    .clientScripts({ content: mockDate });
+```
+
+```json
+{
+    "clientScripts": {
+        "content": "Date.prototype.getTime = () => 42;"
+    }
+}
+```
+
+> You cannot combine the `content`, [path](#specify-the-path-to-a-javascript-file) and [module](#specify-the-module-name) properties.
+
+### Provide Scripts for Particular Pages
+
+You can also specify pages into which a script should be injected. This is helpful when you need to use custom mocks on particular pages and preserve the default behavior everywhere else.
+
+To specify target pages for a script, add the `page` property to the object you pass to `clientScripts`.
+
+*Programming interface*
+
+```text
+runner.clientScripts({ page, path | module | content })
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": { page, path | module | content }
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts({ page, path | module | content })
+test.clientScripts({ page, path | module | content })
+```
+
+Property  | Type                | Description
+--------- | ------------------- | ---------------------------------------------------------------------------
+`page`    | String &#124; RegExp | Specify a page URL to add scripts to a single page, or a regular expression to add scripts to pages whose URLs match this expression.
+
+**Examples**
+
+```js
+runner.clientScripts({
+    page: 'https://myapp.com/page/',
+    path: 'dist/jquery.js'
+});
+```
+
+```json
+{
+    "clientScripts": {
+        "page": "\/user\/profile\/",
```

Comment:
One question: if we support the object notation in the config file, how do we specify RegExps in JSON? @miherlosev

Reply:
Yes. JSON5 library is not supported reqular expressions. So, in configuration we support not all variants due to json5 syntax restrictions.